assets/js/app.js

```diff
@@ -106,6 +114,11 @@ window.addEventListener("phx:dragover-item", (e) => {
   }
 })
 
+window.addEventListener("phx:update-counter", (e) => {
+    const counter = document.querySelector('#counter')
+    Alpine.$data(counter).counter = e.detail.counter
```

Comment:
When Phoenix broadcast an event to the clients we can directly update the Alpine.js component using the `Alpine.$data` function

Reply:
It's not immediately clear to me what the code is doing:
```js
Alpine.$data(counter).counter = e.detail.counter
```
An in-line comment could be helpful to a new learner/reader
e.g: 
```suggestion
    // Lookup the DOM element in the Alpine.$data store and assign the value of e.detail.counter:
    Alpine.$data(counter).counter = e.detail.counter
```